bus-mapping/src/evm/opcodes/callop.rs

```diff
@@ -237,23 +237,35 @@ impl<const N_ARGS: usize> Opcode for CallOpcode<N_ARGS> {
         let code_address = call.code_address();
         match (
             code_address
-                .map(|ref addr| state.is_precompiled(addr))
+                .map(|ref addr| is_precompiled(addr))
                 .unwrap_or(false),
             callee_code_hash.to_fixed_bytes() == *EMPTY_HASH,
         ) {
             // 1. Call to precompiled.
             (true, _) => {
-                warn!("Call to precompiled is left unimplemented");
-
                 for (field, value) in [
                     (CallContextField::LastCalleeId, 0.into()),
                     (CallContextField::LastCalleeReturnDataOffset, 0.into()),
                     (CallContextField::LastCalleeReturnDataLength, 0.into()),
                 ] {
                     state.call_context_write(&mut exec_step, current_call.call_id, field, value);
                 }
+                assert!(call.is_success, "call to precompile should not fail");
+                let caller_ctx = state.caller_ctx_mut()?;
+                let code_address = code_address.unwrap();
+                let (result, contract_gas_cost) = execute_precompiled(
+                    &code_address,
+                    &caller_ctx.memory.0[args_offset..args_offset + args_length],
+                    callee_gas_left,
+                );
+                caller_ctx.memory.0[ret_offset..ret_offset + ret_length]
+                    .copy_from_slice(&result[..ret_length]);
+                for (i, value) in result.iter().enumerate() {
+                    state.memory_write(&mut exec_step, (ret_offset + i).into(), *value)?;
```

Comment:
wrong.

```
                    let caller_id = state.call()?.caller_id;
                    state.push_op(
                        &mut exec_step,
                        RW::WRITE,
                        MemoryOp::new(caller_id, (ret_offset + i).into(), *value)
                    );
```

Reply:
i am not sure whether the fix is good enough... You'd better read the "handle_copy" function of bus-mapping::return_revert.rs. Check anything is missing

Reply:
you also need 

```
                caller_ctx.return_data.resize(ret_length, 0);
                if ret_length != 0 {
                    caller_ctx.return_data[0..length]
                        .copy_from_slice(&result[..ret_length]);
                }
```

Reply:
check https://github.com/scroll-tech/zkevm-circuits/commit/5025861fb8d3ffd5c83022f6c79670d56f6d05a8

Reply:
also 2x rwc_delta in gadget